### ISSUE TITLE
Update / Electron

### ISSUE THREAD

**Issue by jppope** [segmentio/nightmare#267]
I don't know exactly whats going on, still pretty new doing development, but nightmare isn't working, and I'm getting this error below: 

```Uncaught Exception:
TypeError: Error processing argument -1.
    at TypeError (native)
    at NavigationController.loadUrl (/Users/myusername/Desktop/nightmare/node_modules/nightmare/node_modules/electron-prebuilt/dist/Electron.app/Contents/Resources/atom.asar/browser/api/lib/navigation-controller.js:59:24)
    at EventEmitter.webContents.(anonymous function) [as loadUrl] (/Users/myusername/Desktop/nightmare/node_modules/nightmare/node_modules/electron-prebuilt/dist/Electron.app/Contents/Resources/atom.asar/browser/api/lib/web-contents.js:79:27)
    at EventEmitter.<anonymous> (/Users/myusername/Desktop/nightmare/node_modules/nightmare/lib/runner.js:87:23)
    at emitOne (events.js:77:13)
    at EventEmitter.emit (events.js:169:7)
    at process.<anonymous> (/Users/myusername/Desktop/nightmare/node_modules/nightmare/lib/ipc.js:28:10)
    at emitTwo (events.js:87:13)
    at process.emit (events.js:172:7)
    at handleMessage (internal/child_process.js:685:10)```

Everything was fine before I updated... but I can't get the older version back

**Comment by FdezRomero** [comment]
@jppope To get an older version back you can do `npm remove nightmare` and then `npm install nightmare@2.0.5 --save`. Replace `2.0.5` with your desired version.

I also had an error like this when using the old syntax with v2, maybe that's the problem.

**Comment by kanethal** [comment]
I'm having a similar issue. I have a working installation with nightmare 2.0.5. I started a new project and installed the same version, but nightmare doesn't want to initialize. So maybe it's a dependency that is no longer compatible? I only have the one working installation now, and I would just copy it over, but I get the file names too long errors on copy.

**Comment by pastawoua** [comment]
Had an issue with electron from version when they upgraded from 0.33.4 to 0.33.5

I'm maintaining my own fork of nightmare and specified the exact version: https://github.com/pastawoua/nightmare/blob/feature/cammy/package.json

**Comment by tscanlin** [comment]
I can verify this issue as well on both OS X and Window 7. It worked fine with the old version, but I decided I wanted to upgrade and go back to do some refactoring and just can't get it to work at all. Could phantom somehow be interfering? I find it unlikely but not sure of the cause yet, I will keep investigating. 

Here is an error log: 
```
$ DEBUG=nightmare & mocha anonymous.spec.js
[1] 1718


  anonymous users
    Start page
      1) should show form when loaded
TypeError: Cannot read property 'apply' of undefined
    at Nightmare.done (/Users/tim/Code/project_name/node_modules/nightmare/lib/nightmare.js:162:14)
    at Nightmare.next (/Users/tim/Code/project_name/node_modules/nightmare/lib/nightmare.js:140:28)
    at EventEmitter.<anonymous> (/Users/tim/Code/project_name/node_modules/nightmare/lib/nightmare.js:150:12)
    at ChildProcess.<anonymous> (/Users/tim/Code/project_name/node_modules/nightmare/lib/ipc.js:28:10)
    at handleMessage (child_process.js:324:10)
    at Pipe.channel.onread (child_process.js:352:11)
TypeError: Cannot read property 'apply' of undefined
    at Nightmare.done (/Users/tim/Code/project_name/node_modules/nightmare/lib/nightmare.js:162:14)
    at Nightmare.next (/Users/tim/Code/project_name/node_modules/nightmare/lib/nightmare.js:140:28)
    at EventEmitter.<anonymous> (/Users/tim/Code/project_name/node_modules/nightmare/lib/nightmare.js:150:12)
    at ChildProcess.<anonymous> (/Users/tim/Code/project_name/node_modules/nightmare/lib/ipc.js:28:10)
    at handleMessage (child_process.js:324:10)
    at Pipe.channel.onread (child_process.js:352:11)


  0 passing (323ms)
  1 failing

  1) anonymous users Start page should show form when loaded:
     Uncaught TypeError: Cannot read property 'apply' of undefined
      at Nightmare.done (/Users/tim/Code/project_name/node_modules/nightmare/lib/nightmare.js:162:14)
      at Nightmare.next (/Users/tim/Code/project_name/node_modules/nightmare/lib/nightmare.js:140:28)
      at EventEmitter.<anonymous> (/Users/tim/Code/project_name/node_modules/nightmare/lib/nightmare.js:150:12)
      at ChildProcess.<anonymous> (/Users/tim/Code/project_name/node_modules/nightmare/lib/ipc.js:28:10)
      at handleMessage (child_process.js:324:10)
      at Pipe.channel.onread (child_process.js:352:11)



[1]+  Done
```

**Comment by tscanlin** [comment]
Also, for reference, here's the test script I was running:
```
var chai = require('chai');
var Nightmare = require('nightmare');
var expect = chai.expect;

module.exports = function() {
  'use strict';

  describe('anonymous users', function() {
    this.timeout(20000);
    var nightmare;

    before(function() {
      nightmare = new Nightmare({show: true});
    });

    describe('Start page', function () {
      it('should show form when loaded', function (done) {
        new Nightmare()
          .goto('http://yahoo.com')
          .click('.searchsubmit')
          .run();
      });
    });
  });
});
```

I'll post back if I learn anything new.

**Comment by tscanlin** [comment]
I think I figured it out. Looking at this line:
https://github.com/segmentio/nightmare/blob/master/lib/nightmare.js#L162

It looks like the `run` method is expecting a callback, and if you don't pass one in it seems to be causing the error above.

Changing to this fixed it for me, alternatively you should probably pass in a done function from a testing framework for instance, I bet thats the intention with this.

```
.run(function() {
  console.log('DONE')
});
```

Should I open a pull request to at least throw a more helpful error here? Other people might get tripped up by copying/pasting examples online like I did.. :P (Example: http://weblogs.asp.net/mikaelsoderstrom/writing-end-to-end-tests-with-nightmare-js, a top result for using nightmare for testing)

Hope that fixes it for you @jppope!

**Comment by tscanlin** [comment]
Actually the error that you have looks different.. sorry folks!

**Comment by reinpk** [comment]
hey all, i'm not able to reproduce this or really understand exactly what the issue is — if this is still an issue can you summarize the current state/knowledge of the issue?

from my reading of this, you can get rid of `.run()` as that's a `1.x` syntax that isn't needed in `2.x`... see the readme: https://github.com/segmentio/nightmare/blob/master/Readme.md

**Comment by matt-newell** [comment]
Try this... passing the reference of done to the run() method should let the test run properly. 

```var chai = require('chai');
var Nightmare = require('nightmare');
var expect = chai.expect;

module.exports = function() {
  'use strict';

  describe('anonymous users', function() {
    this.timeout(20000);
    var nightmare;

    before(function() {
      nightmare = new Nightmare({show: true});
    });

    describe('Start page', function () {
      it('should show form when loaded', function (done) {
        new Nightmare()
          .goto('http://yahoo.com')
          .click('.searchsubmit')
          .run(**done**);
      });
    });
  });
});```

**Comment by matt-newell** [comment]
Try this... passing the reference of done to the run() method should let the test run properly. 
var chai = require('chai');
var Nightmare = require('nightmare');
var expect = chai.expect;

module.exports = function() {
  'use strict';

  describe('anonymous users', function() {
    this.timeout(20000);
    var nightmare;

    before(function() {
      nightmare = new Nightmare({show: true});
    });

    describe('Start page', function () {
      it('should show form when loaded', function (done) {
        new Nightmare()
          .goto('http://yahoo.com')
          .click('.searchsubmit')
          .run(done);
      });
    });
  });
});
